api/app/Policies/UserPolicy.php

```diff
@@ -118,4 +166,63 @@ protected function applicantHasAppliedToPoolInTeams(User $applicant, $teamIds)
             })
             ->exists();
     }
+
+    /*******************  ROLE CHECKING  *******************/
+
+    /**
+     * Function to check if the acting user can update a team based role
+     *
+     * @return bool
+     */
+    protected function teamAbleToCheck(User $actor, string $roleId, string $teamId)
+    {
```

Comment:
If the user has assign-any-role or assign-any-teamRole should permissions, we can return true right away.

Reply:
Sure https://github.com/GCTC-NTGC/gc-digital-talent/pull/10791/commits/551529946763feb2569114cadf004f3afd4a2412